fix: Properly parse tsconfig.json

zermelo-wisen opened the pull request:
Fixes #91

`tsconfig.json` cannot be parsed with `JSON.parse` properly. It can have trailing comma which is not legal in JSON. Use `typescript.readConfigFile` instead.

dividedmind reviewed (changes requested):
I'm very wary of pulling in whole typescript — it's quite large and will make the recommended usage with _npx_ particularly more inconvenient. I'd look into other ways of parsing it — [tsconfck](https://www.npmjs.com/package/tsconfck) looks promising.

BTW, any change like this should include test cases that didn't work before and now do.

zermelo-wisen commented:
> I'm very wary of pulling in whole typescript — it's quite large and will make the recommended usage with _npx_ particularly more inconvenient. I'd look into other ways of parsing it — [tsconfck](https://www.npmjs.com/package/tsconfck) looks promising.
> 
> BTW, any change like this should include test cases that didn't work before and now do.

I dropped `typescript` and `strip-json-comments` libraries and added `json5` for parsing the `tsconfig.json`.

For the test case, I added a trailing comma to the `tsconfig.json` in `test/typescript-esm`, since tsconfig parsing code path is covered with the `"esm-loader is loaded when required"` integration test in `typescript-esm.test.ts`.

dividedmind reviewed (approved):
Perfect, thank you!

appland-release commented:
:tada: This PR is included in version 2.16.1 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/appmap-node/v/2.16.1)
- [GitHub release](https://github.com/getappmap/appmap-node/releases/tag/v2.16.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
